### PR TITLE
Replace lambda with a helper class

### PR DESCRIPTION
**Pull request by keithc-ca**
The effect is that the VM loads 43 fewer classes to run a 'Hello world' application.

### CODE REVIEW

File: jcl/src/java.base/share/classes/java/lang/ClassLoader.java [hunk 1]
```diff
@@ -116,7 +110,7 @@
   	private Map<String, Boolean> packageAssertionStatus;
   	private Map<String, Boolean> classAssertionStatus;
   	/*[IF Sidecar19-SE]*/
-  	private final Hashtable<String, NamedPackage> packages = new Hashtable();
+    private final Hashtable<String, NamedPackage> packages = new Hashtable<>();
```
Comment (DanHeidinga) [inline jcl/src/java.base/share/classes/java/lang/ClassLoader.java]: tab vs space? Formatting seems off here

File: jcl/src/java.base/share/classes/java/lang/ClassLoader.java [hunk 2]
```diff
@@ -597,13 +596,29 @@ private String checkClassName(String className) {
 }
 
 /*[IF Sidecar19-SE]*/
- /**
-  * Add a class's package name to this classloader's list of packages, if not already present.
+private static final class NamedPackageProvider implements java.util.function.Function<String, NamedPackage> {
```
Comment (DanHeidinga) [inline jcl/src/java.base/share/classes/java/lang/ClassLoader.java]: Can you javadoc this class as well?

File: jcl/src/java.base/share/classes/java/lang/ClassLoader.java (hunk 1 above)
Comment (keithc-ca) [reply jcl/src/java.base/share/classes/java/lang/ClassLoader.java]: The lines before and after have (some) spaces and I assumed it was four of them so I made this line match. Upon closer inspection, the lines above and below start with two spaces and a tab. I suggest any further cleanup should be done separately.

File: jcl/src/java.base/share/classes/java/lang/ClassLoader.java (hunk 2 above)
Comment (keithc-ca) [reply jcl/src/java.base/share/classes/java/lang/ClassLoader.java]: Done.

File: jcl/src/java.base/share/classes/java/lang/ClassLoader.java (hunk 1 above)
Comment (DanHeidinga) [reply jcl/src/java.base/share/classes/java/lang/ClassLoader.java]: Agreed.

**Comment by DanHeidinga** [comment]
Jenkins test sanity plinux jdk11

**Comment by DanHeidinga** [comment]
Jenkins test sanity win jdk8

**Comment by pshipton** [comment]
@keithc-ca Please fix the merge conflict.

**Comment by DanHeidinga** [comment]
Jenkins test sanity plinux jdk11

**Comment by DanHeidinga** [comment]
Jenkins test sanity win jdk8

**Pull request merged**